### PR TITLE
chore: upgrade v4 to use latest contracts

### PR DESCRIPTION
**Pull request by aeolianeth**
Main diff:
- some properties that were previously `bigint` are now just `number`
- Rename `decayRate`, `reservedRate` to `decayPercent`, `reservedPercent`
- Use JBSplit and JBRuleset from SDK where possible
- JBSplit.percent is assumed to be type SplitPortion.

### CODE REVIEW

File: src/packages/v4/components/SplitList/SplitItem/ReservedTokensValue.tsx [hunk 1]
```diff
@@ -4,18 +4,18 @@ import { SplitPortion } from 'juice-sdk-core'
 
 export function ReservedTokensValue({
   splitPercent,
-  reservedRate,
+  reservedPercent,
 }: {
-  splitPercent: bigint
-  reservedRate: number
+  splitPercent: SplitPortion
```
Comment (aeolianeth) [inline src/packages/v4/components/SplitList/SplitItem/ReservedTokensValue.tsx:9]: Note: JBSplit.percent is type SplitPortion

File: src/packages/v4/utils/v4Splits.ts [hunk 2]
```diff
@@ -83,43 +81,42 @@ export const sanitizeSplit = (split: V4Split): V4Split => {
...
-  split2: V4Split
-  split1TotalValue?: bigint
-  split2TotalValue?: bigint
+  split1: JBSplit
+  split2: JBSplit
+  split1TotalValue: bigint
+  split2TotalValue: bigint
 }) {
   const split1Amount = formatWad(
-    (split1TotalValue! * split1.percent) / V4_SPLITS_TOTAL_PERCENT,
```
Comment (aeolianeth) [inline src/packages/v4/utils/v4Splits.ts:116]: note: no `!`

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4CyclesPayoutsPanel/hooks/useV4ExportSplitsToCsv.ts [hunk 3]
```diff
@@ -18,10 +17,10 @@ const CSV_HEADER = [
   'hook',
 ]
 
-const splitToCsvRow = (split: V4Split) => {
+const splitToCsvRow = (split: JBSplit) => {
   return [
     split.beneficiary,
-    `${new SplitPortion(split.percent).formatPercentage()}%`,
+    `${split.percent.formatPercentage()}%`,
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4CyclesPayoutsPanel/hooks/useV4ExportSplitsToCsv.ts:23]: The advantage is that we can do stuff like this.

Comment (johnnyd-eth) [reply src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4CyclesPayoutsPanel/hooks/useV4ExportSplitsToCsv.ts:23]: Very nice

**Review by johnnyd-eth — APPROVED** [review]
Epic stuff, LVGTM

**Pull request merged**